File: backend/server/models/Product.js
const mongoose = require('mongoose');

// 데이터 구조 정의 (Schema)
const productSchema = new mongoose.Schema({
  category: { type: String, required: true }, // 과일, 야채 등
  price: { type: Number, default: 0 },         // 가격
  stock: { type: Number, default: 0 },         // 재고
  name: { type: String, required: true }      // 상품명
});

// 모델로 내보내기 (다른 파일에서 쓸 수 있게 함)
module.exports = mongoose.model('Product', productSchema);

File: backendSpring/src/main/java/com/study/backendSpring/entity/Product.java
package com.study.backendSpring.entity;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import com.fasterxml.jackson.annotation.JsonProperty;

import lombok.Data;
import lombok.Getter;
import lombok.Setter;

@Document(collection = "products") // MongoDB의 컬렉션 이름과 연결
@Data
public class Product {
    @Id
    @JsonProperty("_id")
    private String id;          // MongoDB의 _id 필드
    private String category;    // 과일, 야채 등
    private String name;        // 상품명
    private int price;          // 가격
    private int stock;          // 재고
}